raiden/network/proxies/token_network.py

```diff
@@ -205,18 +206,55 @@ def token_network_deposit_limit(self, block_identifier: BlockSpecification) -> T
             )
         )
 
+    def _token_network_deposit_limit_reached(self, block_identifier: BlockSpecification) -> bool:
+        balance = self.token.balance_of(
+            address=to_checksum_address(self.address), block_identifier=block_identifier
+        )
+        return balance >= self.token_network_deposit_limit(block_identifier=block_identifier)
+
     def safety_deprecation_switch(self, block_identifier: BlockSpecification) -> bool:
         return self.proxy.contract.functions.safety_deprecation_switch().call(
             block_identifier=block_identifier
         )
 
+    def _new_channel_preconditions(
+        self, partner: Address, given_block_identifier: BlockSpecification
+    ):
+        try:
+            channel_already_created = self._channel_exists_and_not_settled(
```

Comment:
this can be replaced by a simple call to `get_channel_identifier` and the function `_new_channel_preconditions` is not necessary